### PR TITLE
Updating teams waterfall skill & fixing cards

### PR DESCRIPTION
**Pull request by Virtual-Josh**
Adding the Teams waterfall dialog skill 
Fixing an issue with the adaptive card scenarios

### CODE REVIEW

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Cards/CardSampleHelper.cs [hunk 1]
```diff
@@ -192,6 +192,21 @@ public static HeroCard CreateHeroCard()
...
 
+        public static HeroCard CreateHeroCard(string type)
+        {
+            var heroCard = new HeroCard
+            {
+                Title = "BotFramework Hero Card",
+                Subtitle = "Microsoft Bot Framework",
+                Text = "Build and connect intelligent bots to interact with your users naturally wherever they are," +
+                       " from text/sms to Skype, Slack, Office 365 mail and other popular services.",
+                Images = new List<CardImage> { new CardImage("https://sec.ch9.ms/ch9/7ff5/e07cfef0-aa3b-40bb-9baa-7c9ef8ff7ff5/buildreactionbotframework_960.jpg") },
```
Comment (gabog) [inline Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Cards/CardSampleHelper.cs:203]: Let's avoid external dependencies, move the image to the project so the tests are self contained.

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Properties/launchSettings.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
+  "$schema": "http://json.schemastore.org/launchsettings.json",
   "iisSettings": {
     "windowsAuthentication": false,
     "anonymousAuthentication": true,
     "iisExpress": {
-      "applicationUrl": "http://localhost:39820/",
+      "applicationUrl": "http://localhost:35420/",
```
Comment (gabog) [inline Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Properties/launchSettings.json:7]: This port is used by the WaterfallSkillBot, let's use a different one for this one (not sure if we will still need it)

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Properties/launchSettings.json [hunk 3]
```diff
@@ -15,13 +16,13 @@
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
     },
-    "TeamsSkillBot": {
+    "WaterfallSkillBotDotNet": {
```
Comment (gabog) [inline Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Properties/launchSettings.json:19]: Wrong name snould be TeamsWaterfallSkillBot

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Bots/SkillBot.cs [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            }
+
+            // This Messaging Extension example allows the user to check a box to include an image with the
+            // shared message.  This demonstrates sending custom parameters along with the message payload.
+            var includeImage = ((JObject)action.Data)["includeImage"]?.ToString();
+            if (!string.IsNullOrEmpty(includeImage) && bool.TrueString == includeImage)
+            {
+                heroCard.Images = new List<CardImage>
+                {
+                    new CardImage { Url = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQtB3AwMUeNoq4gUBGe6Ocj8kyh3bXa9ZbV7u1fVKQoyKFHdkqU" },
```
Comment (gabog) [inline Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Bots/SkillBot.cs:120]: Let's avoid external dependencies, move the image to the project so the tests are self contained.

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Attachments/AttachmentDialog.cs [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+        private static Attachment GetInternetAttachment()
+        {
+            // ContentUrl must be HTTPS.
+            return new Attachment
+            {
+                Name = $@"Files\{Picture}",
+                ContentType = "image/png",
+                ContentUrl = "https://docs.microsoft.com/en-us/bot-framework/media/how-it-works/architecture-resize.png",
```
Comment (gabog) [inline Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Attachments/AttachmentDialog.cs:50]: Let's avoid external dependencies, move the image to the project so the tests are self contained.

**Review by gabog — CHANGES_REQUESTED** [review]
I requested some changes that need to be made, I will continue later.

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Cards/CardSampleHelper.cs (hunk 1 above)
Comment (Virtual-Josh) [reply Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Cards/CardSampleHelper.cs]: consolidated.

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Properties/launchSettings.json (hunk 2 above)
Comment (Virtual-Josh) [reply Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Properties/launchSettings.json]: updated.

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Attachments/AttachmentDialog.cs (hunk 5 above)
Comment (Virtual-Josh) [reply Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/Attachments/AttachmentDialog.cs]: refactored

File: Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Bots/SkillBot.cs (hunk 4 above)
Comment (Virtual-Josh) [reply Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Bots/SkillBot.cs]: resolved.

**Pull request closed without merging**